Flaky test

```
=== RUN   TestNew_withTimeout
2022/04/04 13:57:13 about to sleep
2022/04/04 13:57:13 about to sleep
panic: test timed out after 5s

goroutine 5 [running]:
testing.(*M).startAlarm.func1()
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/testing/testing.go:[178](https://github.com/fnrun/fnrun/runs/5817599997?check_suite_focus=true#step:4:178)8 +0xbb
created by time.goFunc
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/time/sleep.go:[180](https://github.com/fnrun/fnrun/runs/5817599997?check_suite_focus=true#step:4:180) +0x4a

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000089040, {0x71a68d, 0x13}, 0x723fe0)
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/testing/testing.go:1307 +0x752
testing.runTests.func1(0x0)
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/testing/testing.go:1598 +0x9a
testing.tRunner(0xc000089040, 0xc0000c3bf8)
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/testing/testing.go:1259 +0x230
testing.runTests(0xc0000d[208](https://github.com/fnrun/fnrun/runs/5817599997?check_suite_focus=true#step:4:208)0, {0x85f640, 0x9, 0x9}, {0x0, 0xc0000d0640, 0x8635e0})
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/testing/testing.go:1596 +0x7cb
testing.(*M).Run(0xc0000d2080)
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/testing/testing.go:1504 +0x9d2
main.main()
	_testmain.go:61 +0x22c

goroutine 20 [chan receive]:
github.com/fnrun/fnrun/run/fn/pool_test.(*sleepyFn).Invoke(0xc000095d78, {0xc000095cf8, 0x5890b6}, {0x0, 0xc000028800})
	D:/a/fnrun/fnrun/run/fn/pool/pool_test.go:236 +0x85
github.com/fnrun/fnrun/run/fn/pool.(*poolFn).Invoke(0xc0000984c0, {0x761f88, 0xc0000de1e0}, {0x6ec5a0, 0x75e998})
	D:/a/fnrun/fnrun/run/fn/pool/pool.go:40 +0x19d
github.com/fnrun/fnrun/run/fn/pool_test.TestNew_withTimeout(0xc000089520)
	D:/a/fnrun/fnrun/run/fn/pool/pool_test.go:75 +0x29c
testing.tRunner(0xc000089520, 0x723fe0)
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/testing/testing.go:1259 +0x230
created by testing.(*T).Run
	C:/hostedtoolcache/windows/go/1.17.8/x64/src/testing/testing.go:1306 +0x727

goroutine 22 [chan receive]:
github.com/fnrun/fnrun/run/fn/pool_test.(*sleepyFn).Invoke(0xc000041f58, {0xc000041ed8, 0x0}, {0x0, 0x0})
	D:/a/fnrun/fnrun/run/fn/pool/pool_test.go:236 +0x85
github.com/fnrun/fnrun/run/fn/pool.(*poolFn).Invoke(0xc0000984c0, {0x761f88, 0xc0000de1e0}, {0x6ec5a0, 0x75e9a8})
	D:/a/fnrun/fnrun/run/fn/pool/pool.go:40 +0x19d
github.com/fnrun/fnrun/run/fn/pool_test.TestNew_withTimeout.func2()
	D:/a/fnrun/fnrun/run/fn/pool/pool_test.go:66 +0x70
created by github.com/fnrun/fnrun/run/fn/pool_test.TestNew_withTimeout
	D:/a/fnrun/fnrun/run/fn/pool/pool_test.go:65 +0x235
FAIL	github.com/fnrun/fnrun/run/fn/pool	5.042s
```

Reference: https://github.com/fnrun/fnrun/runs/5817599997?check_suite_focus=true

--- a/run/fn/pool/pool_test.go
+++ b/run/fn/pool/pool_test.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"sync"
 	"testing"
 	"time"
 
@@ -57,18 +58,23 @@ func TestNew_withTimeout(t *testing.T) {
 	defer cancel()
 
 	p := newPool(t, "sleepy")
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
 
 	go func() {
+		wg.Done()
 		p.Invoke(ctx, "first input")
 	}()
 
 	go func() {
+		wg.Done()
 		p.Invoke(ctx, "second input")
 	}()
 
 	// The following is cheating a little, but it is to get around a race
 	// race condition with the kicking off of the go routines above.
-	<-time.After(5 * time.Millisecond)
+	wg.Wait()
+	<-time.After(1 * time.Millisecond)
 
 	// pool should be tapped now because the "sleepy" Fn sleeps for a second
 	// before responding

